cmd/docker-notation/docker/signature.go

```diff
@@ -0,0 +1,33 @@
+package docker
+
+import (
+	"context"
+	"net"
+
+	"github.com/distribution/distribution/v3/reference"
+	"github.com/notaryproject/notation-go-lib"
+	"github.com/notaryproject/notation-go-lib/registry"
+	"github.com/notaryproject/notation/pkg/config"
+)
+
+// GetSignatureRepository returns a signature repository
+func GetSignatureRepository(ctx context.Context, ref string) (notation.SignatureRepository, error) {
+	named, err := reference.ParseNamed(ref)
+	if err != nil {
+		return nil, err
+	}
+	hostname, repository := reference.SplitHostname(named)
+
+	tr, err := Transport(hostname)
+	if err != nil {
+		return nil, err
+	}
+
+	insecure := config.IsRegistryInsecure(hostname)
+	if host, _, _ := net.SplitHostPort(hostname); host == "localhost" {
```

Comment:
maybe worth having it as util for determining insecure property from hostname

Reply:
Can we create an issue to track, and merge?